compiler/executable/match_/planner/vertex/constraint.rs

```diff
@@ -133,7 +133,7 @@ impl ConstraintVertex<'_> {
             Self::Links(_) | Self::Has(_) | Self::IndexedRelation(_) => {
                 let unbound_join_variables: Vec<VariableVertexId> = self
                     .variables()
-                    .filter(|&var| self.can_join_on(var) && (!exclude.contains(&var) || include.contains(&var)))
+                    .filter(|&var| self.can_join_on(var) && (!exclude.contains(&var) && include.contains(&var)))
```

Comment:
Replaced the `||` with the `&&` because that felt more correct, but now that I think about it, it's probably not the case. The || Will additionally include steps that are not in the vertex_ordering regardless of whether they are in the ongoing step.

Another question I have is: Does this work for more than 2 constraints being joined?